File: calculatePoints.py
import csv
import json

drivers = {}

with open('Formula1_2025Season_RaceResults.csv', 'r') as f:
    reader = csv.reader(f)
    next(reader) # skip header
    for row in reader:
        if len(row) < 9: continue
        
        # Track,Position,No,Driver,Team... Points is index 8
        # Position is index 1
        pos = row[1].strip()
        no = row[2].strip()
        name = row[3].strip()
        points_str = row[8].strip()
        
        try:
            points = float(points_str)
        except ValueError:
            points = 0
            
        if no not in drivers:
            drivers[no] = {'name': name, 'points': 0, 'podiums': 0}
            
        drivers[no]['points'] += points
        
        if pos in ['1', '2', '3']:
            drivers[no]['podiums'] += 1

with open('points_output.json', 'w', encoding='utf-8') as f:
    json.dump(drivers, f, indent=2)

File: calculatePoints.js
const fs = require('fs');
const content = fs.readFileSync('Formula1_2025Season_RaceResults.csv', 'utf8');
const lines = content.split('\n').filter(l => l.trim().length > 0);
const headers = lines[0].split(',');
// Track,Position,No,Driver,Team... Points is index 8 (0-based)
// Position is index 1

const drivers = {};

for (let i = 1; i < lines.length; i++) {
    const cols = lines[i].split(',');
    if (cols.length < 9) continue;

    const no = cols[2].trim();
    const name = cols[3].trim();
    const pos = cols[1].trim();
    const pointsStr = cols[8].trim();
    const points = parseFloat(pointsStr) || 0;

    // Some lines might be Sprints? The CSV format looks uniform.

    if (!drivers[no]) {
        drivers[no] = {
            name: name,
            points: 0,
            podiums: 0
        };
    }

    drivers[no].points += points;

    if (['1', '2', '3'].includes(pos)) {
        drivers[no].podiums += 1;
    }
}

console.log(JSON.stringify(drivers, null, 2));
